pkg/ui/src/components/NoteButton/NoteButton.tsx

```diff
@@ -0,0 +1,132 @@
+import React from "react";
+
+import ArrowCircleLeft from "@assets/ArrowCircleLeft.svg";
+import ArrowCircleRight from "@assets/ArrowCircleRight.svg";
+
+// #region types
+export interface NoteButtonProps
+  extends React.HTMLAttributes<HTMLButtonElement> {
+  type?: NoteButtonType;
+  label?: NoteButtonLabel | Warehouse;
+  onClick?: () => void;
+}
+
+export enum NoteButtonLabel {
+  RemoveBooks = "Remove Books",
+}
+
+export enum NoteButtonType {
+  In = "in",
+  Out = "out",
+}
+
+export enum Warehouse {
+  Scolastica = "scolastica",
+  Nuovo = "nuovo",
+  Varia = "varia",
+}
+// #endregion types
+
+// #region color
+const colorClassesLookup = {
+  [NoteButtonType.In]: ["text-white", "bg-green-400", "hover:bg-green-500"],
+  [NoteButtonType.Out]: ["text-white", "bg-red-400", "hover:bg-red-500"],
+};
+
+// Classes for focus pseudo-class are the same regardless of color prop
+const focusClasses = [
+  "focus:ring-2",
+  "focus:ring-indigo-500",
+  "focus:ring-offset-2",
+  "focus:ring-offset-white",
+];
+// #endregion color
+
+// #region Component
+/**
+ * A note button component implementing styles from the mockup.
+ * In: green, warehouse label, two side sharp and two side square
+ * Out: red, warehouse label, two side sharp and two side square
+ * Global out: red, remove books label, square
+ * @param {Object} props
+ * @param {NoteButtonType} props.type
+ * @param {NoteButtonLabel | Warehouse} props.label
+ * @param {JSX.Element | JSX.Element[]} props.children
+ */
+const NoteButton: React.FC<NoteButtonProps> = ({
+  children,
+  type = NoteButtonType.In,
+  label = Warehouse.Varia,
+  onClick = () => {},
+  ...props
+}) => {
+  const size = "md";
+  const shape =
+    label === NoteButtonLabel.RemoveBooks ? "rounded-md" : "rounded-r-md";
```

Comment:
kudos 👍 However, I would rather generalise the shape property, but I will update the spec. (my bad)